### PR TITLE
MLPAB-1546: Certificate provider moving from paper to digital

### PR DESCRIPTION
**Pull request by acsauk**
# Purpose

Storing email on CP is likely a temporary step until we decide how we want to handle immaterial changes to an LPA (as the LPA may have already been sent to the LPA store when the CP first logs in). This change at least allows a paper CP to move to online and receive emails.

Fixes MLPAB-1546

### CODE REVIEW

File: internal/page/donor/how_would_certificate_provider_prefer_to_carry_out_their_role.go [hunk 1]
```diff
@@ -58,9 +58,14 @@ type howWouldCertificateProviderPreferToCarryOutTheirRoleForm struct {
 func readHowWouldCertificateProviderPreferToCarryOutTheirRole(r *http.Request) *howWouldCertificateProviderPreferToCarryOutTheirRoleForm {
 	carryOutBy, err := actor.ParseCertificateProviderCarryOutBy(page.PostFormString(r, "carry-out-by"))
 
+	email := page.PostFormString(r, "email")
+	if carryOutBy.IsPaper() {
+		email = ""
+	}
```
Comment (acsauk) [inline internal/page/donor/how_would_certificate_provider_prefer_to_carry_out_their_role.go:64]: I ended up adding this in as we [validate](https://github.com/ministryofjustice/opg-data-lpa-store/blob/main/lambda/create/validate.go#L25) on this field not being present in the lpa store for paper CPs.

File: internal/actor/certificate_provider_provided.go [hunk 2]
```diff
@@ -31,6 +31,8 @@ type CertificateProviderProvidedDetails struct {
 	Tasks CertificateProviderTasks
 	// ContactLanguagePreference is the language the certificate provider prefers to receive notifications in
 	ContactLanguagePreference localize.Lang
+	// Email is the email address returned from OneLogin when the certificate provider logged in
+	Email string
```
Comment (hawx) [inline internal/actor/certificate_provider_provided.go:35]: was this meant to be MLPAB-1546, and so this needs sending to the lpa-store?

Comment (acsauk) [reply internal/actor/certificate_provider_provided.go:35]: My bad - I'd looked at the wrong column when grabbing the ticket number. Updated now.

File: internal/lpastore/update.go [hunk 3]
```diff
@@ -36,7 +36,7 @@ func (c *Client) sendUpdate(ctx context.Context, lpaUID string, actorUID actorui
 	return c.do(ctx, actorUID, req, nil)
 }
 
-func (c *Client) SendCertificateProvider(ctx context.Context, lpaUID string, certificateProvider *actor.CertificateProviderProvidedDetails) error {
+func (c *Client) SendCertificateProvider(ctx context.Context, lpaUID string, certificateProvider *actor.CertificateProviderProvidedDetails, lpa *Lpa) error {
```
Comment (hawx) [inline internal/lpastore/update.go:39]: you could make the signature match `SendAttorney` and remove `lpaUID` here

File: internal/lpastore/update.go [hunk 4]
```diff
@@ -46,27 +46,35 @@ func (c *Client) SendCertificateProvider(ctx context.Context, lpaUID string, cer
...
 	if certificateProvider.HomeAddress.Country != "" {
-		body.Changes = append(body.Changes, updateRequestChange{Key: "/certificateProvider/address/country", New: certificateProvider.HomeAddress.Country})
+		body.Changes = append(body.Changes, updateRequestChange{Key: "/certificateProvider/address/country", New: certificateProvider.HomeAddress.Country, Old: lpa.CertificateProvider.Address.Country})
+	}
+
+	if certificateProvider.Email != "" {
+		body.Changes = append(body.Changes, updateRequestChange{Key: "/certificateProvider/email", New: certificateProvider.Email, Old: lpa.CertificateProvider.Email})
+	}
+
+	if lpa.CertificateProvider.Channel != actor.ChannelOnline {
```
Comment (hawx) [inline internal/lpastore/update.go:76]: I know we only have 2 channels, but I'd prefer `== channel.Paper` here as that is what you are saying it is changing from

**Review by hawx — APPROVED** [review]

**Pull request merged**